internal/customtypes/headers.go

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2025 Ping Identity Corporation
+
+package customtypes
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+var headers []string
+
+type Header struct {
+	Key   string
+	Value string
+}
+
+type HeaderSlice []Header
+
+// Verify that the custom type satisfies the pflag.Value interface
+var _ pflag.Value = (*HeaderSlice)(nil)
+
+func IsValidHeader(header string) (Header, bool) {
+	headerNameRegex := regexp.MustCompile(`(^[^\s]+):[\t ]{0,1}(.*)$`)
+	matches := headerNameRegex.FindStringSubmatch(header)
+	if len(matches) != 3 {
+		return Header{}, false
+	}
+
+	return Header{
+		Key:   matches[1],
+		Value: matches[2],
+	}, true
+}
+
+func (h *HeaderSlice) Set(val string) error {
+	if h == nil {
+		return fmt.Errorf("failed to set Headers value: %s. Headers is nil", val)
+	}
+
+	if val == "" || val == "[]" {
+		return nil
+	} else {
+		valH := strings.SplitSeq(val, ",")
+		for header := range valH {
+			headerVal, isValid := IsValidHeader(header)
+			if !isValid {
+				return fmt.Errorf("failed to set Headers: Invalid header: %s. Headers must be in the proper format", header)
```

Comment:
I think it might be helpful to include the valid regex pattern as part of the message. Thoughts?